scipy/linalg/_decomp_qr.py

```diff
@@ -130,6 +130,11 @@ def qr(a, overwrite_a=False, lwork=None, mode='full', pivoting=False,
         a1 = numpy.asarray(a)
     if len(a1.shape) != 2:
         raise ValueError("expected a 2-D array")
+    
+    # Quick return for square empty array
+    if a1.size == 0:
+        return a1.copy()
```

Comment:
these return arrays and they should be `np.array([])`. I'm typing the same thing so I'll skip the rest but have a look at numpy code for inspiration.